pages/index.js

```diff
@@ -2,79 +2,180 @@ import styled from "styled-components";
-            below.
-          </StyledNoItemsMessage>
+        {/* >>> try to use this: */}
+        {/* {mode === "default" && ( */}
+        {mode !== "edit" && mode !== "add" && (
+          <AddItemContainer>
+            {shoppingItemsWithCategoryColor.length === 0 && (
+              <StyledNoItemsMessage>
+                Your shopping list is empty. Do you need anything?
+              </StyledNoItemsMessage>
+            )}
+            {shoppingItemsWithCategoryColor.length !== 0 &&
+              unpurchasedItems.length === 0 && (
+                <StyledAllItemsPurchasedMessage>
+                  You have got everything on your list. Do you need anything
+                  else?
+                </StyledAllItemsPurchasedMessage>
+              )}
+            <AddButton onClick={() => handleChangeMode("add")}>+</AddButton>
+          </AddItemContainer>
         )}
         {mode === "edit" && (
           <Form
             onSubmitItem={onEditItem}
-            categories={categories}
+            // categories={categories}
             item={editedItem}
-            submitLabel={"update"}
+            // line bvelow not used in form; could be used to adapt the button text
+            // to the button's function in the respective instance of the form
+            // submitLabel={"update"}
             onChangeMode={handleChangeMode}
-            mode={mode}
+            // mode={mode}
           />
         )}
         {mode === "add" && (
           <Form
             onSubmitItem={onAddItem}
-            categories={categories}
-            submitLabel={"submit"}
+            // categories={categories}
+            // submitLabel={"submit"}
             onChangeMode={handleChangeMode}
-            mode={mode}
+            // mode={mode}
           />
         )}
-
-        {mode !== "add" && (
-          <AddItemContainer>
-            <>
-              <p>...need anything else?</p>
+        {mode !== "add" ||
+          (!mode && (
+            <AddItemContainer>
+              <>
+                <p>...need anything else?</p>
+              </>
               <AddButton onClick={() => handleChangeMode("add")}>+</AddButton>
-            </>
-          </AddItemContainer>
+            </AddItemContainer>
+          ))}
+        {unpurchasedItems.length > 0 && (
+          <div>
+            {/* FIX OR DELETE THIS:
+            //  */}
+            {/* <StyledH2>
+              {/* {purchasedItems.length === 0 ? "We need " :  */}
+            {/* Your list contains
+              {unpurchasedItems.length + purchasedItems.length} things.
+            </StyledH2> */}
+
+            <StyledH2>
+              {/* {purchasedItems.length === 0 ? "We need " :  */}
+              There&#39;s {/* } */}
+              {unpurchasedItems.length} thing
+              {unpurchasedItems.length !== 1 ? "s" : ""}{" "}
+              {purchasedItems.length !== 0 ? "left to get" : "on your list"}:
```

Comment:
I understand that you might want to keep the different messages. Please make a decision, update the code accordingly and remove all comments afterwards to keep your code clean. 🧹

Reply:
done earlier